nupic/research/frameworks/pytorch/models/cl_models.py

```diff
@@ -0,0 +1,99 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2019, Numenta, Inc.  Unless you have an agreement
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see http://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+
+from functools import reduce
+
+import torch.nn as nn
+from torch.nn import init
+
+
+class EWCNetwork(nn.Module):
+
+    def __init__(self, input_size, output_size,
+                 hidden_size=400,
+                 hidden_layer_num=2,
+                 hidden_dropout_prob=.5,
+                 input_dropout_prob=.2,
+                 ):
+        """
+        Adapted as is from
+        https://github.com/kuc2477/pytorch-ewc/blob/master/model.py
+
+        Used to test consistency in the algorithm implementation.
+        """
+
+        # Configurations.
+        super().__init__()
+        self.input_size = input_size
+        self.input_dropout_prob = input_dropout_prob
+        self.hidden_size = hidden_size
+        self.hidden_layer_num = hidden_layer_num
+        self.hidden_dropout_prob = hidden_dropout_prob
+        self.output_size = output_size
+
+        # Layers.
+        self.layers = nn.ModuleList([
+            # input
+            nn.Flatten(),
+            nn.Linear(self.input_size, self.hidden_size), nn.ReLU(),
+            nn.Dropout(self.input_dropout_prob),
+            # hidden
+            *((nn.Linear(self.hidden_size, self.hidden_size), nn.ReLU(),
+               nn.Dropout(self.hidden_dropout_prob)) * self.hidden_layer_num),
+            # output
+            nn.Linear(self.hidden_size, self.output_size)
+        ])
+
+        self.xavier_initialize()
+
+    @property
+    def name(self):
+        return (
+            "MLP"
+            "-in{input_size}-out{output_size}"
+            "-h{hidden_size}x{hidden_layer_num}"
+            "-dropout_in{input_dropout_prob}_hidden{hidden_dropout_prob}"
+        ).format(
+            input_size=self.input_size,
+            output_size=self.output_size,
+            hidden_size=self.hidden_size,
+            hidden_layer_num=self.hidden_layer_num,
+            input_dropout_prob=self.input_dropout_prob,
+            hidden_dropout_prob=self.hidden_dropout_prob,
+        )
+
+    def forward(self, x):
+        return reduce(lambda x, l: l(x), self.layers, x)
+
+    def xavier_initialize(self):
+        modules = [
```

Comment:
I'm not opposed to this format, but the string comprehensions are a bit long here. Maybe try:
```
for m in self.modules():
     if not isinstance(m, (nn.Linear, nn.Conv2d)):
          continue
     for p in m.parameters():
          if p.dim() >= 2:
               init.xavier_normal(p)
```

Reply:
Same as above

Reply:
Nevermind about this, I don't think mine is any better.